### ISSUE TITLE
Add yes and no callbacks

### ISSUE THREAD

**Issue by jcalixte** [caroso1222/notyf#11]
Hi,

First of all, thank you for this dead simple lib.

I just have one request, is it possible to add buttons et callback options to confirm and alert methods?

Thank you

Julien

**Comment by toish** [comment]
Personally I think this runs against the purpose of this library, sounds like you want more of a dialog then a notification. Check out [Vex](https://github.com/hubspot/vex) or [Alertify](http://alertifyjs.com/) for super-custom dialogs

**Comment by jcalixte** [comment]
Well, I have to agree with you, this library do one thing and it do it right.
Thank you for your refs!

**Comment by caroso1222** [comment]
Thanks both for your input on this issue! 

It's a great idea to have those buttons and callbacks, but I have to agree with @toish in that Notyf's mission is to serve as a pretty simple, minimal library to show simple notifications.

I'm going to close this issue but I'm happy to reopen it if someone else brings back the discussion.

Thanks again for your input and sorry for being so silent lately.

**Comment by luongvanphuc** [comment]
@caroso1222 I've recently replaced toastr with notyf, and found that notyf doesn't have callback options, so I had to come back with toastr.

Personally I think the buttons may be not necessary but the callbacks do. Sometime we want to redirect user to somewhere after clicking on the noti.

I raise my idea here because my company app is really big and the bundle size is a big problem. It's best if we can use tiny libs for big apps.

In the contrary, I also agree with @jcalixte that "this library does one thing and does it right", so it's all up to you.

**Comment by caroso1222** [comment]
Thanks for your input @luongvanphuc. I've just opened a PR adding Notyf events. You'll now be able to listen to 'click' and 'dismiss' events.

```javascript
const notyf = new Notyf();
notyf.success('Foo bar')
     .on('click', () => console.log('I was clicked');
```

On the other hand, Yes/No callbacks can be implemented using custom HTML in the message content.

**Comment by kith-and-kin** [comment]
> Thanks for your input @luongvanphuc. I've just opened a PR adding Notyf events. You'll now be able to listen to 'click' and 'dismiss' events.
> 
> ```js
> const notyf = new Notyf();
> notyf.success('Foo bar')
>      .on('click', () => console.log('I was clicked');
> ```
> 
> On the other hand, Yes/No callbacks can be implemented using custom HTML in the message content.

This would be great. I'd love a callback functionality on the dismiss button especially, so we can save whether a message has been dismissed by way of cookie.

**Comment by caroso1222** [comment]
@kith-and-kin yeah, that functionality is on the [PR](https://github.com/caroso1222/notyf/pull/74/files#diff-04c6e90faac2675aa89e2176d2eec7d8R137-R150) too. I'm just waiting for Friday to review my PR with a fresh brain. It'll be released on v3.8. Thanks for your input!
